packages/shopware-6-client/src/services/customerService.ts

```diff
@@ -0,0 +1,168 @@
+import {
+  getCustomerEndpoint,
+  getCustomerAddressEndpoint,
+  getCustomerUpdateEmailEndpoint,
+  getCustomerUpdatePasswordEndpoint,
+  getCustomerDefaultBillingAddressEndpoint,
+  getCustomerDefaultShippingAddressEndpoint
+} from "../endpoints";
+import { Customer } from "../interfaces/models/checkout/customer/Customer";
+import { apiService } from "../apiService";
+import { CustomerAddress } from "../interfaces/models/checkout/customer/CustomerAddress";
+import { CustomerRegistrationParams } from "../interfaces/request/CustomerRegistrationParams";
+import { update } from "..";
+import { UpdateContextResponse } from "../interfaces/response/UpdateContextResponse";
+
+interface CustomerRegisterResponse {
+  data: string;
+}
+
+/**
+ * Register a customer
+ */
+export async function register(
+  params: CustomerRegistrationParams
+): Promise<CustomerRegisterResponse> {
+  const resp = await apiService.post(getCustomerEndpoint(), params);
+  return resp.data;
+}
+
+interface CustomerLoginParam {
+  username: string;
+  password: string;
+}
+
+/**
+ * Get the context token for current user
+ */
+export async function login(
+  params: CustomerLoginParam
+): Promise<UpdateContextResponse> {
+  const resp = await apiService.post(`${getCustomerEndpoint()}/login`, params);
+  const contextToken = resp.data["sw-context-token"];
+  update({ contextToken });
+  return { contextToken };
+}
+
+/**
+ * End up the session
+ */
+export async function logout(): Promise<void> {
+  await apiService.post(`${getCustomerEndpoint()}/logout`);
```

Comment:
Possible to use `getCustomerLogoutEndpoint()`. :)

Reply:
sure, fixed